tools/spanner/purge_ttl.py

```diff
@@ -41,39 +42,103 @@ def from_env():
+        if len(args.collection_ids) == 1:
+            query += " = {:d}".format(args.collection_ids[0])
+        else:
+            query += " in ({})".format(
+                ', '.join(map(str, args.collection_ids)))
+    return query
+
+
+def spanner_purge(args, request=None):
+    instance = client.instance(args.instance_id)
+    database = instance.database(args.database_id)
 
-    logging.info("For {}:{}".format(instance_id, database_id))
+    logging.info("For {}:{}".format(args.instance_id, args.database_id))
+    batch_query = add_conditions(
+            args,
+            'DELETE FROM batches WHERE expiry < CURRENT_TIMESTAMP()'
+        )
+    bso_query = add_conditions(
+            args,
+            'DELETE FROM bsos WHERE expiry < CURRENT_TIMESTAMP()'
+        )
 
     # Delete Batches. Also deletes child batch_bsos rows (INTERLEAVE
     # IN PARENT batches ON DELETE CASCADE)
-    with statsd.timer("syncstorage.purge_ttl.batches_duration"):
-        batches_start = datetime.now()
-        query = 'DELETE FROM batches WHERE expiry < CURRENT_TIMESTAMP()'
-        result = database.execute_partitioned_dml(query)
-        batches_end = datetime.now()
-        logging.info("batches: removed {} rows, batches_duration: {}".format(
-            result, batches_end - batches_start))
 
+    deleter(
+        database,
+        name="batches",
+        query=batch_query
+    )
     # Delete BSOs
-    with statsd.timer("syncstorage.purge_ttl.bso_duration"):
-        bso_start = datetime.now()
-        query = 'DELETE FROM bsos WHERE expiry < CURRENT_TIMESTAMP()'
-        result = database.execute_partitioned_dml(query)
-        bso_end = datetime.now()
-        logging.info("bso: removed {} rows, bso_duration: {}".format(
-            result, bso_end - bso_start))
+    deleter(
+        database,
+        name="bso",
+        query=bso_query
+    )
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description="Purge old TTLs"
+    )
+    parser.add_argument(
+        "-i",
+        "--instance_id",
+        default=os.environ.get("INSTANCE_ID", "spanner-test"),
+        help="Spanner instance ID"
+    )
+    parser.add_argument(
+        "-d",
+        "--database_id",
+        default=os.environ.get("DATABASE_ID", "sync_schema3"),
+        help="Spanner Database ID"
+    )
+    parser.add_argument(
+        "-u",
+        "--sync_database_url",
+        default=os.environ.get("SYNC_DATABASE_URL"),
+        help="Spanner Database DSN"
+    )
+    parser.add_argument(
+        "--collection_ids",
+        default=os.environ.get("COLLECTION_IDS", "[]"),
+        help="JSON array of collection IDs to purge"
+    )
+    args = parser.parse_args()
+    collections = json.loads(args.collection_ids)
+    if not isinstance(collections, list):
+        collections = [collections]
+    args.collection_ids = collections
+    if args.sync_database_url and not (
+            args.instance_id and args.database_id):
```

Comment:
`INSTANCE/DATABASE_ID` have a default so they'll pretty much always override a `SYNC_DATABASE_URL` here. I think we can kill these defaults (or maybe these args entirely unless @erkolson or anyone else uses them, I don't), [the cron job uses `SYNC_DATABASE_URL`](https://github.com/mozilla-services/cloudops-infra/blob/ea0adbe/projects/sync/k8s/charts/sync/templates/ttl-purge-cron.yaml#L49).

Reply:
I have always used the `SYNC_DATABASE_URL` and was unaware of the other options.

Reply:
Heh, I use the others because I can never remember what the spanner DSN is. I've reworked things a bit to be smarter and more clear what's going on. Thanks!